conans/test/functional/toolchains/test_nmake_toolchain.py

```diff
@@ -21,11 +29,32 @@ def test_toolchain_nmake(compiler, version, runtime, cppstd, build_type):
                 "build_type": build_type,
                 "arch": "x86_64"}
 
+    serialize_array = lambda arr: "[{}]".format(",".join([f"'{v}'" for v in arr]))
+    conf = {
+        "tools.build:defines": serialize_array(defines) if defines else "",
+        "tools.build:cflags": serialize_array(cflags) if cflags else "",
+        "tools.build:cxxflags": serialize_array(cxxflags) if cxxflags else "",
+        "tools.build:sharedlinkflags": serialize_array(sharedlinkflags) if sharedlinkflags else "",
+        "tools.build:exelinkflags": serialize_array(exelinkflags) if exelinkflags else "",
+    }
+
     # Build the profile according to the settings provided
     settings = " ".join('-s %s="%s"' % (k, v) for k, v in settings.items() if v)
+    conf = " ".join(f'-c {k}="{v}"' for k, v in conf.items() if v)
     client.run("new dep/1.0 -m=cmake_lib")
-    client.run(f'create . -tf=None {settings} '
-               f'-c tools.cmake.cmaketoolchain:generator="Visual Studio 15"')
+    client.run(f'create . -tf=None {settings} {conf}')
+
+    # Rearrange defines to macro / value dict
+    conf_preprocessors = {}
+    for define in defines:
+        if "=" in define:
+            key, value = define.split("=", 1)
+            if not value:
+                conf_preprocessors[key] = "1"
+            else:
+                conf_preprocessors[key] = value
+        else:
+            conf_preprocessors[define] = "1"
```

Comment:
instead of re-arranging, provide the defines already in a dict, and compose it into list when necessary

Reply:
`@pytest.mark.parametrize` cannot take dict values AFAIK. Do you have an example?

Reply:
```python
[{"TEST_DEFINITION1": None}, {"TEST_DEFINITION2": "0"}, .........],
```

Reply:
I think I've tried and it didn't work. I can try again.

Reply:
Actually it's not robust enough. If I try to inject `/DTEST_DEFINITION1` for {"TEST_DEFINITION1": None} as an input, it won't be understand like this by check_exe_run. `/DTEST_DEFINITION1` is equivelent to `/DTEST_DEFINITION1=1`, but I also want to test that `/DTEST_DEFINITION` AND `/DTEST_DEFINITION=1` works in `NMakeToolchain`.